aquality-tracking-cucumber5-jvm/src/main/java/aquality/tracking/integrations/cucumber5jvm/TestCaseNameParser.java

```diff
@@ -0,0 +1,51 @@
+package aquality.tracking.integrations.cucumber5jvm;
+
+import io.cucumber.core.internal.gherkin.ast.Feature;
+import io.cucumber.core.internal.gherkin.ast.ScenarioOutline;
+import io.cucumber.core.internal.gherkin.ast.TableRow;
+import io.cucumber.plugin.event.TestCase;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+class TestCaseNameParser {
+
+    private static final String SCENARIO_OUTLINE_KEYWORD = "Scenario Outline";
+
+    private final Feature feature;
+    private final TestCase testCase;
+
+    TestCaseNameParser(Feature feature, TestCase testCase) {
+        this.feature = feature;
+        this.testCase = testCase;
+    }
+
+    public String getScenarioName() {
+        String scenarioName = testCase.getName();
+        if (testCase.getKeyword().equals(SCENARIO_OUTLINE_KEYWORD)) {
+            scenarioName = getScenarioOutlineName(scenarioName);
+        }
+        return feature.getName().concat(scenarioName);
+    }
+
+    private String getScenarioOutlineName(final String scenarioName) {
+        List<TableRow> examplesTableRows = feature.getChildren().stream()
+                .filter(child -> child.getName().equals(scenarioName))
+                .map(node -> (ScenarioOutline) node)
+                .map(outline -> outline.getExamples().get(0))
+                .flatMap(examples -> examples.getTableBody().stream())
+                .collect(Collectors.toList());
+
+        int tableRowIndex = IntStream.range(0, examplesTableRows.size())
+                .filter(i -> examplesTableRows.get(i).getLocation().getLine() == testCase.getLine())
+                .findFirst()
+                .orElse(-1);
+
+        String scenarioOutlineName = "";
```

Comment:
maybe instead of empty string it is better to have constant like "SCENARIO_NAME_IS_UNDEFINED"?

Reply:
Actually no, because this validation will be on Cucumber side, and it Scenario Outline doesn't have examples it won't be compiled. So, it will never be an empty string.

Reply:
Not sure that it will occur ever, because Cucumber should handle it. But updated it a little bit.